### PR TITLE
[nativeaot][tests] Add Native AOT iOS perf scenarios

### PR DESCRIPTION
**Pull request by kotlarmilos**
This PR updates the CI jobs to include the Native AOT iOS performance scenarios. The changes include building the NativeAOT iOS scenarios and running them, as well as updating the underlying scripts and project files to support it. Additionally, it separates the scenarios from the current Mono AOT sample measurements.

Additionally, it temporarily fixes regression to allow testing. The regression will be fixed on the main in a separate PR tracked in https://github.com/dotnet/runtime/issues/80911.

Preliminary perf measurements for the HelloiOS app using Native AOT.

SOD - iOS HelloWorld Native AOT Zip Size
Metric                                           |Average           |Min               |Max
-------------------------------------------------|------------------|------------------|------------------
SOD - iOS HelloWorld Native AOT Zip Size         |1727305.000 bytes |1727305.000 bytes |1727305.000 bytes

SOD - iOS HelloWorld Native AOT .app
Metric                                       |Average           |Min               |Max
---------------------------------------------|------------------|------------------|------------------
SOD - iOS HelloWorld Native AOT .app         |8075485.000 bytes |8075485.000 bytes |8075485.000 bytes
pub/nativeaot/HelloiOS.app/HelloiOS          |8074496.000 bytes |8074496.000 bytes |8074496.000 bytes
pub/nativeaot/HelloiOS.app/Info.plist        |981.000 bytes     |981.000 bytes     |981.000 bytes
pub/nativeaot/HelloiOS.app/PkgInfo           |8.000 bytes       |8.000 bytes       |8.000 bytes

Device Startup - iOS HelloWorld Native AOT
Metric          |Average        |Min            |Max
----------------|---------------|---------------|---------------
Generic Startup |38.600 ms      |34.000 ms      |41.000 ms

### CODE REVIEW

**Comment by kotlarmilos** [comment]
Blocked by https://github.com/dotnet/performance/issues/3010.

**Comment by ivanpovazan** [comment]
FWIW, it would be great to also have the SOD measurements/comparisons with symbols stripped out

**Comment by kotlarmilos** [comment]
Good point. I suggest we make symbol stripping and extraction the default behavior for both Mono and Native AOT samples. Later, in the dashboard there could be an option to include symbols in the size measurements.

File: src/tasks/AppleAppBuilder/Templates/CMakeLists.txt.template
Comment (ivanpovazan) [reply src/tasks/AppleAppBuilder/Templates/CMakeLists.txt.template:87]: Thanks, I opened a draft PR that references these and similar required changes: https://github.com/dotnet/runtime/pull/86652

File: src/mono/sample/iOS/Makefile
Comment (ivanpovazan) [inline src/mono/sample/iOS/Makefile:9]: I would prefer that the https://github.com/dotnet/runtime/pull/85317/files#diff-425d33169424e0e20bc3bc87ca31fffa5764c9d9eefdcd59f3f6e2c4902da30eR83 specifies explicitly `STRIP_DEBUG_SYMBOLS=true` rather than to have it set by default. 
The reason for this is that stripping symbols also obfuscates the signature and shouldn't be the default behaviour. It is true that symbol stripping is on by default with Xamarin Release builds, but Xamarin also has additional post processing tasks that reapply the signature on a stripped app, making the final app deployable on a device.

Additionally, it might worth considering having 4 measurements:
- MonoAOT with symbols
- MonoAOT without symbols
- NativeAOT with symbols
- NativeAOT without symbols

Although, it might not need to be part of this PR (just a thought).

File: src/mono/sample/iOS-NativeAOT/Makefile
Comment (ivanpovazan) [inline src/mono/sample/iOS-NativeAOT/Makefile:9]: Same comment as for MonoAOT: https://github.com/dotnet/runtime/pull/85317/files#diff-8f074c3f5f920c745799be4983690457f3b036bc566a65d5644b95568ba4b34dL9

File: src/mono/sample/iOS/Makefile
Comment (ivanpovazan) [reply src/mono/sample/iOS/Makefile:9]: Regarding:
> Additionally, it might worth considering having 4 measurements:

There is a TODO task in https://github.com/dotnet/runtime/issues/80905:
> [ ] Include measuring HelloiOS application size without debug symbols

which should/would be covered by this.

Comment (kotlarmilos) [reply src/mono/sample/iOS/Makefile:9]: Sounds good! There will be the following measurements:
 - MonoAOT llvm with symbols
 - MonoAOT llvm without symbols
 - MonoAOT nollvm with symbols
 - MonoAOT nollvm without symbols
 - NativeAOT with symbols
 - NativeAOT without symbols

**Comment by kotlarmilos** [comment]
This PR is ready for review. The performance results can be found in the `dotnet-runtime-perf` pipeline. The Mono performance SOD runs have been expanded to include separate runs with and without symbols. Once this PR is merged, the following job names need to be updated in the dashboard.

Native AOT performance measurements:
 - SOD - iOS HelloWorld Native AOT .app Size symbols
 - SOD - iOS HelloWorld Native AOT .app Size nosymbols
 - SOD - iOS HelloWorld Native AOT Zip Size symbols
 - SOD - iOS HelloWorld Native AOT Zip Size nosymbols
 - SOD - Device Startup - iOS Native AOT HelloWorld symbols
 - SOD - Device Startup - iOS Native AOT HelloWorld nosymbols
 
Mono performance measurements:
- SOD - iOS HelloWorld Mono .app Size llvm symbols
- SOD - iOS HelloWorld Mono .app Size llvm nosymbols
- SOD - iOS HelloWorld Mono .app Size nollvm symbols
- SOD - iOS HelloWorld Mono .app Size nollvm nosymbols
- SOD - iOS HelloWorld Mono Zip Size llvm symbols
- SOD - iOS HelloWorld Mono Zip Size llvm nosymbols
- SOD - iOS HelloWorld Mono Zip Size nollvm symbols
- SOD - iOS HelloWorld Mono Zip Size nollvm nosymbols
- Device Startup - iOS Mono HelloWorld llvm symbols
- Device Startup - iOS Mono HelloWorld llvm nosymbols
- Device Startup - iOS Mono HelloWorld nollvm symbols
- Device Startup - iOS Mono HelloWorld nollvm nosymbols

@LoopedBard3 I will keep you posted.

File: src/tasks/AppleAppBuilder/Templates/CMakeLists.txt.template
Comment (ivanpovazan) [reply src/tasks/AppleAppBuilder/Templates/CMakeLists.txt.template:87]: Feel free to include these in this PR, so it does not block it from getting upstreamed. 
The PR mentioned in my previous comment will adapt this further and should not affect the behaviour.

File: src/mono/sample/iOS-NativeAOT/Program.csproj
Comment (ivanpovazan) [inline src/mono/sample/iOS-NativeAOT/Program.csproj:32]: Please take a look at my comment: https://github.com/dotnet/runtime/pull/85317/files#r1210389112

File: src/tasks/AppleAppBuilder/Templates/CMakeLists.txt.template
Comment (kotlarmilos) [reply src/tasks/AppleAppBuilder/Templates/CMakeLists.txt.template:87]: Thanks for the info.

**Review by ivanpovazan — APPROVED** [review]
LGTM! 
Just a small note on "symbol-stripped" scenarios, that in those cases we are stripping local symbols only via `strip -x`

**Comment by kotlarmilos** [comment]
> LGTM! Just a small note on "symbol-stripped" scenarios, that in those cases we are stripping local symbols only via `strip -x`

Thanks! If we maintain consistency across all platforms and strip in the same way, it should be fine. If I recall correctly, we encountered an issue with global symbols stripping.

**Pull request merged**
